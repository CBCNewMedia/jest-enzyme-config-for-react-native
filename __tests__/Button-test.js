/**
 * @format
 */

import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import Button from '../Button';

// Note: this is just for use with Jest snapshot testing
// and comes packaged with react-native init project.
// You do not need this if using Enzyme 'toMatchSnapshot' etc.
import renderer from 'react-test-renderer';

// This test just uses Jest snapshot testing
it('renders correctly, test using Jest', () => {
  renderer.create(<Button />);
});


// Using Jest + Enzyme
describe('<Button />', () => {
  it('renders correctly, test using Jest + Enzyme', () => {
    expect(shallow(<Button/>)).toMatchSnapshot();
  });

  describe('onPressHandler', () => {
    it('should call onPress', () => {
        // Arrange
        const mockOnPress = jest.fn();      // 1. mock function
        const component = shallow(<Button
            label= "test label"
            onPress={mockOnPress}           // 2. passing in mock function as props
        />);
        const instance = component.instance();  // 3. getting an instance of component

        // Act
        instance.onPressHandler();          // 4. manually triggering onPressHandler()

        // Assert
        expect(mockOnPress).toHaveBeenCalled();
        expect(mockOnPress).toHaveBeenCalledTimes(1);   // 5. checking return values
    });
  });
});
