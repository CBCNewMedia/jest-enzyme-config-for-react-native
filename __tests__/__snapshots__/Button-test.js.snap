// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders correctly, test using Jest + Enzyme 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Object {
      "alignSelf": "stretch",
      "backgroundColor": "#38ba7d",
      "borderBottomColor": "#1e6343",
      "borderBottomWidth": 6,
      "borderWidth": 1,
      "height": 45,
      "justifyContent": "center",
      "marginLeft": 15,
      "marginRight": 15,
    }
  }
>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "color": "#fff",
        "fontSize": 16,
        "fontWeight": "600",
      }
    }
  />
</TouchableOpacity>
`;
